numpyro/distributions/distribution.py

```diff
@@ -9,16 +9,83 @@
 import numpy as onp
 import scipy.stats as osp_stats
 from scipy._lib._util import getargspec_no_self
-from scipy.stats._distn_infrastructure import instancemethod
+from scipy.stats._distn_infrastructure import instancemethod, rv_frozen, rv_generic
 
 import jax.numpy as np
 from jax import device_put, lax
 from jax.numpy.lax_numpy import _promote_args
 from jax.random import _is_prng_key
 from jax.scipy import stats
 
+from numpyro.distributions.transforms import AffineTransform
+
+
+class jax_frozen(rv_frozen):
+    _validate_args = False
+
+    def __init__(self, dist, *args, **kwargs):
+        self.args = args
+        self.kwds = kwargs
+
+        # create a new instance
+        self.dist = dist.__class__(**dist._updated_ctor_param())
+
+        shapes, _, scale = self.dist._parse_args(*args, **kwargs)
+        if self._validate_args:
+            # TODO: check more concretely for each shape parameter
+            if not np.all(self.dist._argcheck(*shapes)):
+                raise ValueError('Invalid shape parameters provided to the distribution.')
+            if not np.all(scale > 0):
+                raise ValueError('Invalid scale parameter provided to the distribution.')
+
+        self.a, self.b = self.dist.a, self.dist.b
+
+    @property
+    def support(self):
+        return self.dist._support(*self.args, **self.kwds)
+
+    def __call__(self, size=None, random_state=None):
```

Comment:
Could you explain why this is needed?

Reply:
This is no need given that we have patched scipy implementation. But doing this way, we don't need to patch scipy. I intentionally put it here to get your notice. ;)